core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiService.scala

```diff
@@ -65,6 +66,27 @@ trait WorkspaceApiService extends UserInfoDirectives {
           }
         }
     } ~
+      path("workspaces" / "tags") {
+        parameter('q.?) { queryString =>
+          get {
+            complete {
+              workspaceServiceConstructor(userInfo).getTags(queryString)
+            }
+          }
+        }
+      } ~
+      path("workspaces" / Segment ) { workspaceId =>
+        get {
+          parameterSeq { allParams =>
+            traceRequest { span =>
+              complete {
+                workspaceServiceConstructor(userInfo).getWorkspaceById(workspaceId,
+                  WorkspaceFieldSpecs.fromQueryParams(allParams, "fields"), span)
+              }
+            }
+          }
+        }
+      } ~
```

Comment:
the `path("workspaces" / "tags")` had to move higher, so that it is considered first; otherwise, `path("workspaces" / Segment )` would match it.

As a weird complication of these APIs having the same shape: an end user could, in the swagger UI, enter "tags" as the UUID for the get-workspace-by-id API, and it will return the tags payload. Do we care about this?

Reply:
Agreed tags isn't actually a uuid, but I think we should avoid reusing the namespace. Since we're already using the namespace for workspace namespace, there's already potential for conflicts between 'tags' and the workspace namespace, and I think we should avoid the possibility of conflicts between uuid and namespace. The meaning I would infer from this exact path: "workspaces"/{variable} would be getting all workspaces for a workspace namespace, since we're already using workspaces/{namespace}/{workspaceName}. I would vote for something like workspaces/id/{workspaceId}

Reply:
yeah, unfortunately because `workspaces/{namespace}/{workspaceName}` exists, using `workspaces/id/{workspaceId}` translates to making the string "id" unusable as a namespace. blech, I will sleep on this!

Reply:
after some research and realizing that namespaces should always be 6+ characters, I agree with your suggestion and have changed the api signature to `/workspaces/id/{workspaceId}`!